scripts/externalTests/test_helpers.py

```diff
@@ -0,0 +1,284 @@
+        if self.solc_binary_type == "solcjs":
+            # TODO: add support to solc-js # pylint: disable=fixme
+            raise NotImplementedError()
+        print("Setting up solc...")
+        solc_version_output = subprocess.getoutput(f"{self.solc_binary_path} --version").split(":")[1]
+        return parse_solc_version(solc_version_output)
+
+    @staticmethod
+    def on_local_test_dir(fn):
+        """Run a function inside the test directory"""
+
+        def f(self, *args, **kwargs):
+            assert self.test_dir is not None
+            os.chdir(self.test_dir)
+            return fn(self, *args, **kwargs)
+
+        return f
+
+    def setup_environment(self):
+        """Configure the project build environment"""
+        print("Configuring Runner building environment...")
+        replace_version_pragmas(self.test_dir)
+
+    @on_local_test_dir
+    def clean(self):
+        """Clean temporary directories"""
+        rmtree(self.tmp_dir)
+
+    @on_local_test_dir
+    def compiler_settings(self, _: List[str]):
+        # TODO: default to hardhat # pylint: disable=fixme
+        raise NotImplementedError()
+
+    @on_local_test_dir
+    def compile(self, _: str):
+        # TODO: default to hardhat # pylint: disable=fixme
+        raise NotImplementedError()
+
+    @on_local_test_dir
+    def run_test(self):
+        # TODO: default to hardhat # pylint: disable=fixme
+        raise NotImplementedError()
+
+
+# Helper functions
+def compiler_settings(evm_version: str, via_ir: str = "false", optimizer: str = "false", yul: str = "false") -> dict:
+    return {
+        "optimizer": {"enabled": optimizer, "details": {"yul": yul}},
+        "evmVersion": evm_version,
+        "viaIR": via_ir,
+    }
+
+
+def settings_from_preset(preset: str, evm_version: str) -> dict:
+    assert preset in AVAILABLE_PRESETS
+    switch = {
+        "legacy-no-optimize": compiler_settings(evm_version),
+        "ir-no-optimize": compiler_settings(evm_version, via_ir="true"),
+        "legacy-optimize-evm-only": compiler_settings(evm_version, optimizer="true"),
+        "ir-optimize-evm-only": compiler_settings(evm_version, via_ir="true", optimizer="true"),
+        "legacy-optimize-evm+yul": compiler_settings(evm_version, optimizer="true", yul="true"),
+        "ir-optimize-evm+yul": compiler_settings(evm_version, via_ir="true", optimizer="true", yul="true"),
+    }
+    assert preset in switch
+    return switch[preset]
+
+
+def parse_command_line(description: str, args: List[str]):
+    arg_parser = ArgumentParser(description)
+    arg_parser.add_argument(
+        "solc_binary_type",
+        metavar="solc-binary-type",
+        type=str,
+        default="native",
+        choices=["native", "solcjs"],
+        help="""Solidity compiler binary type""",
+    )
+    arg_parser.add_argument(
+        "solc_binary_path",
+        metavar="solc-binary-path",
+        type=Path,
+        default=Path("/usr/local/bin/solc"),
```

Comment:
I don't think this is a good default. It's not universal - for example on my system it's `/usr/bin/solc` instead. Also, even if you have that binary, it's not necessarily the one you want to use. Very often you'll want to use the one you just built. I think it's better if you have explicitly specify it. Otherwise it's too easy not to realize you're using the wrong one.

If you want to make it convenient, I'd instead make the script accept relative paths so that you can just write `prb-math.py native solc`.

Reply:
Right. I just removed the default value.